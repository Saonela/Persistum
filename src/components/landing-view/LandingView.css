
.landing-wrapper {
    overflow: auto;
    height: 100%;
    width: 100%;
}

.landing {
    display: flex;
    flex-wrap: wrap;
    position: relative;
    max-width: 875px;
    height: calc(100% - 200px);
    padding: 75px var(--padding-l) var(--padding-l);
    margin: 10vh auto 0 auto;
    text-align: left;
}

.landing__content {
    z-index: 0;
    padding: 0 16px;
}

.landing__right {
    z-index: 0;
    border: 1px solid;
    width: 340px;
    margin-left: -287px;
    margin-top: 162px;
}

.landing__header {
    margin: 0;
    margin-bottom: 42px;
    color: var(--color-text);
    font-size: var(--font-size-xl);
}

.landing__header span {
    color: var(--color-theme);
}

.landing__description {
    font-size: 18px;
    line-height: 30px;
    padding-top: 50px;
}

.landing__footer {
    display: flex;
    align-items: center;
    justify-content: space-around;
    width: 100%;
    height: 80px;
    margin-top: auto;
    margin-bottom: calc(var(--padding-l) * -1);
    border-top: 1px solid;
    font-size: 13px;
}

.landing__get-started {
    display: flex;
    position: relative;
    width: 352px;
    margin-top: -8px;
    margin-bottom: 76px;
    font-size: 16px;
    align-items: center;
    justify-content: space-between;
}

.main-button,
.secondary-button {
    height: 40px;
    width: 148px;
    font-size: 18px;
    font-family: Lato, sans-serif;
    background: var(--color-theme);
    font-weight: bold;
    border-radius: 8px;
    box-shadow: 12px 8px 31px -12px rgba(43, 44, 52, 0.6);
    color: white;
    outline: 0;
    border: none;
    cursor: pointer;
    transition: opacity 200ms ease-out, box-shadow 200ms ease-out;
    opacity: 1;
}

.secondary-button {
    border: 1px solid var(--color-theme);
    background: #ffffff;
    color: var(--color-theme);
}

.main-button:hover,
.secondary-button:hover {
    opacity: 0.8;
}

.main-button:hover,
.secondary-button:hover,
.main-button:focus,
.secondary-button:focus {
    box-shadow: 12px 8px 31px -6px rgba(43, 44, 52, 0.6);
}

.main-illustration {
    display: flex;
    align-items: center;
    position: relative;
    top: -122px;
    left: 360px;
    width: 356px;
}

.main-illustration__list,
.main-illustration__stats,
.main-illustration__calendar {
    width: 100px;
}

.main-illustration__calendar {
    margin-left: 14px;
}

.main-illustration__stats {
    margin-left: 25px;
}

.figures-group {
    position: absolute;
    left: 50vw;
    transform: translate(-900px, -23px) rotate(-47deg);
}

.background-figure {
    background: #4973d3;
    border-radius: 100%;
    box-shadow: 41px 18px 121px -63px #4973d3;
}

.background-figure-1 {
    position: absolute;
    width: 2000px;
    height: 2000px;
    top: -1260px;
    left: 81%;
    opacity: 0.9;
}

.background-figure-2 {
    width: 150px;
    height: 150px;
    top: 393px;
    left: -24px;
    opacity: 0.6;
}

.background-figure-3 {
    width: 175px;
    height: 175px;
    top: 213px;
    left: 50px;
    opacity: 0.7;
}

.background-figure-4 {
    width: 200px;
    height: 200px;
    top: 11px;
    left: 136px;
    opacity: 0.8;
}

@media only screen and (max-width: 900px) {
    .landing {
        margin-top: 5vh;
    }

    .landing__content {
        margin-left: 7vw;
    }

    .landing__header {
        font-size: 48px;
    }

    .main-illustration {
        flex-direction: column;
        top: -178px;
        left: 50vw;
        width: 128px;
    }

    .main-illustration *:not(:last-child) {
        margin-bottom: 24px;
    }

    .main-illustration__list {
        margin-left: 20px;
    }

    .landing__get-started {
        top: -222px;
    }
}

@media only screen and (max-width: 720px) {
    .landing {
        justify-content: center;
        text-align: center;
    }

    .landing__content {
        margin-left: 0;
    }

    .landing__description {
        padding-top: 12px;
    }

    .main-illustration {
        flex-direction: row;
        justify-content: space-between;
        top: 0;
        left: 0;
        width: auto;
        margin-top: 75px;
    }

    .main-illustration *:not(:last-child) {
        margin-bottom: 0;
    }

    .main-illustration__list {
        margin-left: 0;
    }

    .landing__get-started {
        justify-content: space-evenly;
        top: 0;
        width: 100%;
        margin-top: 80px;
        margin-bottom: 76px;
    }
}

@media only screen and (max-width: 500px) {
    .landing__description br {
        display: none;
    }

    .landing__get-started {
        flex-direction: column;
        height: 180px;
    }

    .main-illustration {
        flex-direction: column;
        margin-top: 50px;
    }

    .main-illustration *:not(:last-child) {
        margin-bottom: 24px;
    }

    .main-illustration * {
        margin-left: 0;
    }

    .main-illustration__list {
        margin-left: 3px;
    }
}

@media only screen and (max-width: 400px) {
    .landing__description br {
        display: none;
    }
}
