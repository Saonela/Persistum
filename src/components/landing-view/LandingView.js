import React, {useEffect, useState} from "react";
import "./LandingView.css"
import list from "../../assets/landing/list.svg"
import calendar from "../../assets/landing/calendar.svg"
import stats from "../../assets/landing/stats.svg"
import {Link, withRouter} from "react-router-dom";
import AuthAPIService from "../../services/api/authAPIService";

function LandingView({history, onReturningUser, onLoadingStateChange}) {

    const [clicked, setClicked] = useState(false);

    useEffect(() => {
        if (clicked) {
            onLoadingStateChange(true);
            AuthAPIService.getCurrentUser().then((user) => {
                if (user) {
                    history.push('/form');
                    onReturningUser();
                } else {
                    history.push('/register');
                }
                onLoadingStateChange(false);
            });
        }
    }, [clicked]);

    return (
        <React.Fragment>
            <div className="landing-wrapper">
                <div className="background-figure background-figure-1"/>
                <div className="figures-group">
                    <div className="background-figure background-figure-2"/>
                    <div className="background-figure background-figure-3"/>
                    <div className="background-figure background-figure-4"/>
                </div>
                <div className="landing">
                    <div className="landing__content">
                        <h1 className="landing__header">Welcome to <span>Persistum.</span></h1>
                        <p className="landing__description">
                            Never let your routine get out of order. <br/>
                            Easy way to organize your tasks or goals. <br/>
                            Log and track your activities. <br/>
                            Overview your progress. <br/>
                            And keep going to achieve your goals!</p>
                        <figure className="main-illustration">
                            <img className="main-illustration__list" src={list} alt="list-illustration"/>
                            <img className="main-illustration__calendar" src={calendar} alt="calendar-illustration"/>
                            <img className="main-illustration__stats" src={stats} alt="stats-illustration"/>
                        </figure>
                        <section className="landing__get-started">
                            <button className="main-button" onClick={() => setClicked(true)}>Get started</button>
                            <Link to="/login" tabIndex="-1">
                                <button className="secondary-button">Login</button>
                            </Link>
                        </section>
                    </div>
                    <footer className="landing__footer">
                        <a href="mailto:&#107;&#118;&#097;&#108;&#105;&#110;&#116;&#101;&#108;&#105;&#115;&#064;&#103;&#109;&#097;&#105;&#108;&#046;&#099;&#111;&#109;">kvalintelis@gmail.com</a>
                        <p>2020</p>
                    </footer>
                </div>
            </div>
        </React.Fragment>
    )
}

export default withRouter(LandingView);
