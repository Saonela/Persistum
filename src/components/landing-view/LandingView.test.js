import React from 'react';
import {render} from "@testing-library/react";
import {fireEvent} from "@testing-library/dom";
import { Router } from 'react-router-dom'
import { createMemoryHistory } from 'history'
import LandingView from "./LandingView";

const mockState = {};

jest.mock('react-redux', () => ({
    useDispatch: jest.fn(() => {}),
    useSelector: jest.fn((callback) => {
        return callback(mockState)}),
}));

describe('LandingView', () => {

    beforeEach(() => {
        Object.assign(mockState, {user: null});
    });

    it('should go to links', () => {
        const history = createMemoryHistory()
        history.push('/');
        const {getByText} = render(<Router history={history}><LandingView/></Router>)
        const registerButton = getByText('Get started');
        const loginButton = getByText('Login');
        fireEvent.click(registerButton);
        expect(history.location.pathname).toBe('/register');
        fireEvent.click(loginButton);
        expect(history.location.pathname).toBe('/login');
    });

    it('should change register link to app link if logged in', () => {
        Object.assign(mockState, {user: {email: 'test@test'}});
        const history = createMemoryHistory()
        history.push('/');
        const {getByText} = render(<Router history={history}><LandingView/></Router>)
        const registerButton = getByText('Get started');
        fireEvent.click(registerButton);
        expect(history.location.pathname).toBe('/form');
    });

    it('should match snapshot', () => {
        const history = createMemoryHistory()
        const {asFragment} = render(<Router history={history}><LandingView/></Router>);
        expect(asFragment()).toMatchSnapshot();
    });

});
