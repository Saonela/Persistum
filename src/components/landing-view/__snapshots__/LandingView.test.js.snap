// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingView should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="landing-wrapper"
  >
    <div
      class="background-figure background-figure-1"
    />
    <div
      class="figures-group"
    >
      <div
        class="background-figure background-figure-2"
      />
      <div
        class="background-figure background-figure-3"
      />
      <div
        class="background-figure background-figure-4"
      />
    </div>
    <div
      class="landing"
    >
      <div
        class="landing__content"
      >
        <h1
          class="landing__header"
        >
          Welcome to 
          <span>
            Persistum.
          </span>
        </h1>
        <p
          class="landing__description"
        >
          Never let your routine get out of order. 
          <br />
          Easy way to organize your tasks or goals. 
          <br />
          Log and track your activities. 
          <br />
          Overview your progress. 
          <br />
          And keep going to achieve your goals!
        </p>
        <figure
          class="main-illustration"
        >
          <img
            alt="list-illustration"
            class="main-illustration__list"
            src="list.svg"
          />
          <img
            alt="calendar-illustration"
            class="main-illustration__calendar"
            src="calendar.svg"
          />
          <img
            alt="stats-illustration"
            class="main-illustration__stats"
            src="stats.svg"
          />
        </figure>
        <section
          class="landing__get-started"
        >
          <a
            href="/register"
            tabindex="-1"
          >
            <button
              class="main-button"
            >
              Get started
            </button>
          </a>
           or
          <a
            href="/login"
            tabindex="-1"
          >
            <button
              class="secondary-button"
            >
              Login
            </button>
          </a>
        </section>
      </div>
      <footer
        class="landing__footer"
      >
        <a
          href="mailto:kvalintelis@gmail.com"
        >
          kvalintelis@gmail.com
        </a>
        <p>
          2020
        </p>
      </footer>
    </div>
  </div>
</DocumentFragment>
`;
